Sources/SwiftFormat/Core/Parsing.swift

```diff
@@ -25,22 +25,36 @@ import SwiftSyntax
 ///
 /// - Parameters:
 ///   - source: The Swift source code to be formatted.
+///   - operatorTable: The operator table to use for sequence folding.
 ///   - url: A file URL denoting the filename/path that should be assumed for this syntax tree,
 ///     which is associated with any diagnostics emitted during formatting. If this is nil, a
 ///     dummy value will be used.
-///   - operatorTable: The operator table to use for sequence folding.
+///   - experimentalFeatures: The set of experimental features that should be enabled in the parser.
+///     The names of these features correspond to the names of the `Parser.ExperimentalFeatures`
+///     enum in the `SwiftParser` module of swift-syntax.
 ///   - parsingDiagnosticHandler: An optional callback that will be notified if there are any
 ///     errors when parsing the source code.
 /// - Throws: If an unrecoverable error occurs when formatting the code.
 func parseAndEmitDiagnostics(
   source: String,
   operatorTable: OperatorTable,
   assumingFileURL url: URL?,
+  experimentalFeatures: Set<String>,
   parsingDiagnosticHandler: ((Diagnostic, SourceLocation) -> Void)? = nil
 ) throws -> SourceFileSyntax {
-  let sourceFile =
-    operatorTable.foldAll(Parser.parse(source: source)) { _ in }.as(SourceFileSyntax.self)!
-
+  var experimentalFeaturesSet: Parser.ExperimentalFeatures = []
+  for featureName in experimentalFeatures {
+    guard let featureValue = Parser.ExperimentalFeatures(name: featureName) else {
+      throw SwiftFormatError.unrecognizedExperimentalFeature(featureName)
+    }
+    experimentalFeaturesSet.formUnion(featureValue)
+  }
+  var source = source
+  let sourceFile = source.withUTF8 { sourceBytes in
+    operatorTable.foldAll(Parser.parse(source: sourceBytes, experimentalFeatures: experimentalFeaturesSet)) { _ in }.as(
+      SourceFileSyntax.self
+    )!
```

Comment:
```suggestion
    operatorTable.foldAll(Parser.parse(source: sourceBytes, experimentalFeatures: experimentalFeaturesSet)) { _ in }
    .as(SourceFileSyntax.self)!
```

Reads better, I think.

Reply:
Done.